packages/foundry/script/DeployHelpers.s.sol

```diff
@@ -15,20 +15,37 @@ contract ScaffoldETHDeploy is Script {
   string root;
   string path;
   Deployment[] public deployments;
+  uint256 constant ANVIL_FIRST_PK =
+    0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80;
+  address constant ANVIL_FIRST_ACCOUNT =
+    0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266;
+
+  function _startBroadcast() internal returns (address deployer) {
+    vm.startBroadcast();
+    (, deployer,) = vm.readCallers();
 
-  function setupLocalhostEnv() internal returns (uint256 localhostPrivateKey) {
     if (block.chainid == 31337) {
-      root = vm.projectRoot();
-      path = string.concat(root, "/localhost.json");
-      string memory json = vm.readFile(path);
-      bytes memory mnemonicBytes = vm.parseJson(json, ".wallet.mnemonic");
-      string memory mnemonic = abi.decode(mnemonicBytes, (string));
-      return vm.deriveKey(mnemonic, 0);
-    } else {
-      return vm.envUint("DEPLOYER_PRIVATE_KEY");
+      // starts a broadcast and reads the caller
+      vm.stopBroadcast();
+      // check balance of first anvil account
+      // if balance is not 0, is the first run
+      uint256 balance = ANVIL_FIRST_ACCOUNT.balance;
+      if (balance > 1 ether && deployer != vm.addr(ANVIL_FIRST_PK)) {
```

Comment:
Maybe we could update this condition as 

`if(balance == 0 && vm.addr(ANVIL_FIRST_PK))`

and while sending eth maybe just send 500 ETH?, Also lets declare 500 ETH as constant above so if anyone want to change he can update it

Reply:
i have simplified the logic by using only `balance == 0` because if your account is the `LAST_ANVIL_ACCOUNT` , it is supposed to have funds, otherwise everything else ( even the local faucet on script would fail )